### ISSUE TITLE
System.InvalidOperationException: Connection is not open when using npgsql

### ISSUE THREAD

**Issue by alekw** [NLog/NLog#5547]
**NLog version**: 5.3.2

**Platform**:  .NET8

**Current NLog config**

```xml
<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://www.nlog-project.org/schemas/NLog.xsd NLog.xsd"
      autoReload="true"
      throwExceptions="true"
      internalLogLevel="Off">

  <targets>
    <target name="PostgresqlLogger" xsi:type="Database"
            dbProvider="Npgsql.NpgsqlConnection, Npgsql"
            keepConnection="true"
            connectionString="Server=localhost;Database=database;User Id=postgres;Password=postgres;"
            commandType="Text"
            commandText="INSERT INTO logs(MachineName, MessageBody) 
          values(@MachineName, @MessageBody>
      <parameter name="@MachineName" layout="${machinename}"/>
      <parameter name="@MessageBody" layout="${message}"/>
    </target>
  </targets>

  <rules>
    <logger name="*" minlevel="Trace" writeTo="PostgresqlLogger" final="true" />
  </rules>
</nlog>


```

- What is the current result?
After updating from nlog 4.6.8 to 5.3.2 I am getting occasionally the following exception:
Description: The process was terminated due to an unhandled exception.
Exception Info: System.InvalidOperationException: Connection is not open

- What is the expected result?
Lack of exception, as it was before.
- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?
In progress.
- Please post full exception details (message, stacktrace, inner exceptions)
Description: The process was terminated due to an unhandled exception.
Exception Info: System.InvalidOperationException: Connection is not open
   at Npgsql.ThrowHelper.ThrowInvalidOperationException(String message)
   at Npgsql.NpgsqlCommand.ExecuteReader(Boolean async, CommandBehavior behavior, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteNonQuery(Boolean async, CancellationToken cancellationToken)
   at NLog.Targets.DatabaseTarget.ExecuteDbCommandWithParameters(LogEventInfo logEvent, IDbConnection dbConnection, IDbTransaction dbTransaction)
   at NLog.Targets.DatabaseTarget.WriteLogEventSuppressTransactionScope(LogEventInfo logEvent, String connectionString)
   at NLog.Targets.DatabaseTarget.Write(LogEventInfo logEvent)
   at NLog.Targets.Target.Write(AsyncLogEventInfo logEvent)
   at NLog.Targets.Target.WriteAsyncThreadSafe(AsyncLogEventInfo logEvent)
   at NLog.Targets.Target.WriteAsyncLogEvent(AsyncLogEventInfo logEvent)
   at NLog.LoggerImpl.Write(Type loggerType, TargetWithFilterChain targetsForLevel, LogEventInfo logEvent, LogFactory logFactory)
   at NLog.Logger.WriteToTargets(LogEventInfo logEvent, ITargetWithFilterChain targetsForLevel)
   ...my code...
   at System.Threading.ExecutionContext.RunForThreadPoolUnsafe[TState](ExecutionContext executionContext, Action`1 callback, TState& state)
   at System.Threading.QueueUserWorkItemCallback`1.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
- Are there any workarounds? no
- Is there a version in which it did work? 4.6.8
- Can you help us by writing an unit test? N/A

**Comment by snakefoot** [comment]
Think you have added `throwExceptions="true"` to perform troubleshooting, but forgot removing it again.

`throwExceptions="true"` is mostly for internal NLog unit-testing, and should never be used in production environment.

But it is recommended to use `throwConfigExceptions="true"` for validation of NLog configuration.

**Comment by snakefoot** [comment]
Closing issue as resolved, due to inactivity.

**Comment by snakefoot** [comment]
Maybe the just released NLog v5.3.3 will allow you use `throwExceptions="true"` together with the NLog Database-Target:
- https://www.nuget.org/packages/NLog/5.3.3

**Comment by alekw** [comment]
Thanks, I will give it a try.
